### ISSUE TITLE
Feature Request: Trigger build after pull request is closed

### ISSUE THREAD

**Issue by catto** [screwdriver-cd/screwdriver#1707]
**What happened**:
Some users deploy their apps every open pull request with `~pr` trigger and the apps would be named with each pull request number.
Once a pull request is merged or closed, no one takes care of temporarily deployed apps from pull request build. Then unused/abandoned apps retain on their infrastructure.

**What you expected to happen**:
Screwdriver triggers a build after each pull request is closed so that users can run teardown build and keep their compute cluster clean.

For example,
```
jobs:
    pullrequest:
        requires: ~pr
        image: node:8
        steps:
        - test: build and test
        - deploy: <deploy app using SD_PULL_REQUEST environment variable as its name>
        - check: <test against http://endpoint-pr-$SD_PULL_REQUEST.example.com>
    pr-teardown:
        # triggers after pr is closed
        requires: ~prclosed
        image: node:8
        steps:
        - teardown: <remove endpoint-pr-$SD_PULL_REQUEST.example.com app>
```

**Comment by jithin1987** [comment]
How is this different than enabling pull request chain and user adding a job to do clean up ? Even betterm this teardown can be combined with a template and handled in same PR job.

**Comment by catto** [comment]
@jithin1987 @jithin1987 Users can deploy, test, teardown in a single event using chainPR feature. But in some cases, users don't want to teardown their instance automatically after tests. 

For instance, screwdriver has [homepage](https://github.com/screwdriver-cd/homepage). Each author has to get screenshot and paste it to the PR. (https://github.com/screwdriver-cd/homepage/pull/41 ).

If there is PR-closed trigger, we can achieve a workflow like below: 
* When new PR is open, deploy the homepage to a temporary instance without teardown job.
* The deploy job adds a comment like "changes have been deployed to http://homepage-pr-X.example.com" to the PR. 
* We can check if the changes works properly on each reviewer's browsers/environment. 
* The instance will be cleaned up after PR is closed.
